### PR TITLE
Switch to using Boto3

### PR DESCRIPTION
**Pull request by Frozenball**

### CODE REVIEW

File: pontus/amazon_s3_file_validator.py [hunk 1]
```diff
@@ -50,14 +51,19 @@ class AmazonS3FileValidator(object):
...
-        function that takes Boto S3 Key instance as a parameter.
+        function that takes Boto S3 Object instance as a parameter.
 
     """
     def __init__(self, key_name, bucket, validators=[]):
         self.errors = []
-        self.key = bucket.get_key(key_name)
-        if not self.key:
-            raise FileNotFoundError(key=key_name)
+        self.key = bucket.Object(key_name)
```
Comment (vesauimonen) [inline pontus/amazon_s3_file_validator.py]: Should we rename `key` to `object` or `obj`? The terminology change seems to be `Key` -> `Object`, `key_name` -> `key`.

File: pontus/amazon_s3_signed_request.py [hunk 2]
```diff
@@ -74,12 +74,14 @@ def __init__(
...
         mime_type,
         bucket,
+        session,
         acl='public-read',
         expires_in=60,
         success_action_status='201',
         max_content_length=None,
         randomize=False,
     ):
+        # bucket.meta.client._request_signer._credentials
```
Comment (vesauimonen) [inline pontus/amazon_s3_signed_request.py]: remove comment

File: tests/test_amazon_s3_file_validator.py [hunk 3]
```diff
@@ -12,16 +13,15 @@
 
 class CustomValidator(BaseValidator):
     def __call__(self, key):
-        if key.name != 'test-unvalidated-uploads/images/hello.jpg':
+        if key.key != 'test-unvalidated-uploads/images/hello.jpg':
```
Comment (vesauimonen) [inline tests/test_amazon_s3_file_validator.py]: `key` -> `obj`?

**Comment by Frozenball** [comment]
I renamed key to obj

**Comment by vesauimonen** [comment]
@Frozenball fix tests and we're good to go!

**Comment by Frozenball** [comment]
Oh I didn't notice the tests were broken 😅

**Pull request merged**